.github/workflows/json_schema.yaml

```diff
@@ -0,0 +1,41 @@
+name: JSON schema validation
+on:
+  pull_request: {}
+
+jobs:
+  validate:
+    name: Validate values.schema.json with schemalint
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "^1.19"
+
+      - name: Install schemalint
+        run: |
+          go install github.com/giantswarm/schemalint@latest
+
+      - name: Run schemalint
+        run: |
+          schemalint verify ./helm/cluster-azure/values.schema.json --rule-set cluster-app
+  generate:
+    name: Check that values.yaml is generated from values.schema.json with helm-values-gen
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: actions/setup-go@v3
+        with:
+          go-version: "^1.19"
+
+      - name: Install helm-values-gen
+        run: |
+          go install github.com/giantswarm/helm-values-gen@latest
+
+      - name: Run helm-values-gen
+        run: |
+          diff ./helm/cluster-azure/values.yaml <(helm-values-gen helm/cluster-azure/values.schema.json)
```

Comment:
If there is a difference, the only output will be a diff, correct? I think a more speaking message would be ideal.

Reply:
Yes, you can see an example of a failing out [here](https://github.com/giantswarm/cluster-azure/actions/runs/4240682862/jobs/7369980385).

You're right. Let me check how to do that

Reply:
I have added some output below the diff. You can see the results [here](https://github.com/giantswarm/cluster-azure/actions/runs/4241353322/jobs/7371461481).

I don't think that I can do it better inside this job. If we want more control over the output, we could add more functionality to helm values gen.
Something like: `helm-values-gen validate values.schema.json values.yaml`